pymongo/cursor.py

```diff
@@ -144,6 +144,11 @@ def __init__(self, collection, filter=None, projection=None, skip=0,
         if not isinstance(limit, int):
             raise TypeError("limit must be an instance of int")
         validate_boolean("no_cursor_timeout", no_cursor_timeout)
+        if no_cursor_timeout:
```

Comment:
suggest: `if no_cursor_timeout and not self.__explicit_session:`

This way users can avoid seeing the warning by applying the workaround described here: https://docs.mongodb.com/v4.4/reference/method/cursor.noCursorTimeout/#session-idle-timeout-overrides-nocursortimeout

Reply:
Thank you for pointing this out!